File: backend/app/schemas/analytics.py
"""Response contracts for Feature 6 (Analytics) endpoints."""

from typing import List

from pydantic import BaseModel


class ActivityTrendPoint(BaseModel):
    date: str  # "YYYY-MM-DD"
    count: int


class ActivityTrendResponse(BaseModel):
    points: List[ActivityTrendPoint]


class ResponseRateResponse(BaseModel):
    submitted: int
    responded: int
    rate: float  # 0-100

File: frontend/src/types/analytics.ts
// Mirrors backend/app/schemas/analytics.py.

export interface ActivityTrendPoint {
  date: string; // "YYYY-MM-DD"
  count: number;
}

export interface ActivityTrendResponse {
  points: ActivityTrendPoint[];
}

export interface ResponseRateResponse {
  submitted: number;
  responded: number;
  rate: number; // 0-100
}
